clients/cobol-lsp-vscode-extension/src/__tests__/ZoweApiTest.ts

```diff
@@ -12,25 +12,131 @@
+        await expect(zoweApi.fetchMember("dataSet", "theMember", "profile")).resolves.toEqual("content");
+        expect(getExpectStringMock).toBeCalledTimes(1);
+        expect(getExpectStringMock.mock.calls[0][1]).toBe("/zosmf/restfiles/ds/dataSet(theMember)");
+        expect(createSessionMock).toBeCalledWith("profile");
+    });
+    it("listMember call RestClient with right path", async () => {
+        const getExpectJsonMock = jest.fn().mockResolvedValue({
+            items: [
+                {member: "member1"},
+                {member: "member2"},
+            ],
+        });
+        RestClient.getExpectJSON = getExpectJsonMock;
+        const zoweApi = new ZoweApi();
+        const createSessionMock = jest.fn().mockResolvedValue(new Session());
+        zoweApi.createSession = createSessionMock;
+        await expect(zoweApi.listMembers("dataSet", "profile")).resolves.toStrictEqual(["member1", "member2"]);
+        expect(createSessionMock).toBeCalledWith("profile");
+        expect(getExpectJsonMock).toBeCalledTimes(1);
+        expect(getExpectJsonMock.mock.calls[0][1]).toBe("/zosmf/restfiles/ds/dataSet/member")
+    });
     it("throw No Password error if profile don't have a password", async () => {
-        (BasicProfileManager as any).load = jest.fn().mockResolvedValue({profile: {password: ""}});
+        loadMock.mockResolvedValue({profile: {password: ""}});
         const zoweApi = new ZoweApi();
         const expectedError = new ZoweError("No password", Type.NoPassword);
-        expect(zoweApi.createSession("zoweProfile")).rejects.toEqual(expectedError);
+        await expect(zoweApi.createSession("zoweProfile")).rejects.toEqual(expectedError);
+        expect(loadMock).toBeCalledWith({name: "zoweProfile"});
     });
     it("createSession works fine if password is set", async () => {
-        (BasicProfileManager as any).load = jest.fn().mockResolvedValue({profile: {password: "secret"}});
+        loadMock.mockResolvedValue({profile: {password: "secret", host: "theHost"}});
+        const zoweApi = new ZoweApi();
+        const expectedSession = {
+            hostname: "theHost",
+            password: "secret",
+            type: "basic",
+        };
+        expect(zoweApi.createSession("zoweProfile")).resolves.toEqual({ISession: expectedSession});
+        expect(loadMock).toBeCalledWith({name: "zoweProfile"});
+    });
+    it("Creates profile params with ZOWE_CLI_HOME env variable", () => {
+        const OLD_ZOWE_CLI_HOME = process.env.ZOWE_CLI_HOME;
+        process.env.ZOWE_CLI_HOME = "zoweHome";
+        const zoweApi = new ZoweApi();
+        const expectedParams = {
+            profileRootDirectory: path.join("zoweHome", "profiles"),
+            type: "zosmf",
+        };
+        try {
+            expect((zoweApi as any).createProfileParams()).toEqual(expectedParams)
+        } finally {
+            process.env.ZOWE_CLI_HOME = OLD_ZOWE_CLI_HOME;
+        }
```

Comment:
did you consider to have a ```setup()``` and ```tearDown()``` to initialize and rollback the env variable?
Since seems important to restore the env var at the end of the test I prefer to have it in a method to avoid any problem if in the future we'll refactor this test and accidentally remove the finally clause.

Reply:
Good point. I'll try to rewrite it.